contracts/NFTController.sol

```diff
@@ -1,10 +1,48 @@
 // SPDX-License-Identifier: MIT
 
 pragma solidity ^0.8.0;
+import "./Mocks/IERC721.sol";
 
+contract NFTController {
+IERC721 public nonFungibleContract;
```

Comment:
so with this as a global here, it would allow anyone to call `TokenAuction` and change the NFT being auctioned, which could brick the contract. we should remove the `TokenAuction` function and let the user pass in the NFT address to `createAuction` and store that address in the `Auction` struct